internal/alertmanager/currentalert/process.go

```diff
@@ -239,47 +239,47 @@ func addPvc(log logr.Logger, alertLabels model.LabelSet, alertAnnotations model.
 }
 
 func resizePvc(log logr.Logger, labels model.LabelSet, annotiations model.LabelSet, client client.Client) error {
-	cr, err := k8sutil.GetCr(string(labels["kafka_cr"]), string(labels["namespace"]), client)
+
+	pvc, err := getPvc(string(labels["persistentvolumeclaim"]), string(labels["namespace"]), client)
 	if err != nil {
 		return err
 	}
 
-	pvc, err := getPvc(string(labels["persistentvolumeclaim"]), string(labels["namespace"]), client)
+	cr, err := k8sutil.GetCr(pvc.Labels["kafka_cr"], string(labels["namespace"]), client)
 	if err != nil {
 		return err
 	}
 
 	for i, broker := range cr.Spec.Brokers {
 		if strconv.Itoa(int(broker.Id)) == pvc.Labels["brokerId"] {
-			var storageConfigs []v1beta1.StorageConfig
-
-			if broker.BrokerConfig == nil {
-				storageConfigs = cr.Spec.BrokerConfigGroups[cr.Spec.Brokers[i].BrokerConfigGroup].StorageConfigs
-			} else {
-				storageConfigs = broker.BrokerConfig.StorageConfigs
+			brokerConfig, err := util.GetBrokerConfig(broker, cr.Spec)
+			if err != nil {
+				return errors.WrapIf(err, "failed to determine broker config")
 			}
 
-			for _, c := range storageConfigs {
-				if c.MountPath == pvc.Annotations["mountPath"] {
-					size := *c.PvcSpec.Resources.Requests.Storage()
+			storageConfigs := brokerConfig.StorageConfigs
+
+			for n, c := range storageConfigs {
+				modifiableConfig := c.DeepCopy()
+				if modifiableConfig.MountPath == pvc.Annotations["mountPath"] {
+					size := *modifiableConfig.PvcSpec.Resources.Requests.Storage()
 
 					incrementBy, err := resource.ParseQuantity(string(annotiations["incrementBy"]))
```

Comment:
Move this to the beginning of the `resizePvc` method instead of re-reading it in every step of the loop